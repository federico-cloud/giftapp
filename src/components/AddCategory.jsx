import React, { useState } from 'react';

export const AddCategory = ({setCategorias}) => {
  
    const [inputValue, setInputValue] = useState('Escriba aqui su categoria');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategorias( cats => [inputValue,...cats]);
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}
