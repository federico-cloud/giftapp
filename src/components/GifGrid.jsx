import React, { useEffect, useState } from 'react'

export const GifGrid = ({categoria}) => {

    const [imagenes, setImagenes] = useState([]);

    useEffect( () => {
        getGifs();
    }, [])

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=goku&limit=20&api_key=FQNhGq9VGKl9uLcBgROf6bKArM2jKtTR';
        const resp = await fetch(url);

        const {data} = await resp.json();

        const gifs = data.map( img => {
            return{
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        setImagenes(gifs);
    } 
  
    return (
        <>
            <h3>{categoria}</h3>
            <ol>
                {
                    imagenes.map( img => (
                        <li key={img.id}> {img.title}</li>
                    ))
                }
            </ol>
        </>
    )
}
