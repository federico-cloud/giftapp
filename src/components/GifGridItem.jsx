import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ( { title ,url } ) => {

    return (
        <div className="card animate__animated animate__zoomIn">
            <p>{title}</p>
            <img src={url} alt={title}/>
        </div>
    )

}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}