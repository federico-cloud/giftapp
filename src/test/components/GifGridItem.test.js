import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Titulo';
    const url = 'htpps://localhost/algo.jpg'

    test('Debe mostrar el componente de manera correctamente', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url} />);

        expect(wrapper).toMatchSnapshot();
    })

})