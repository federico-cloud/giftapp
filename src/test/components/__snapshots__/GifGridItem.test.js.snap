// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente de manera correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <p>
    Titulo
  </p>
  <img
    alt="Titulo"
    src="htpps://localhost/algo.jpg"
  />
</div>
`;
